### PR TITLE
update to using elasticsearch api v6.2

### PR DESCRIPTION
**Pull request by maksim-lauryk**

### CODE REVIEW

File: src/main/groovy/com/epam/esp/jira/issue/IssueIndexer.groovy [hunk 1]
```diff
@@ -95,9 +101,14 @@ class IssueIndexer implements Runnable {
...
-                logMessage += ' got result from API '
-                logMessage += elasticSearchHelper.updateItem(issue.key, item)
+
+                def issueJson = new JsonSlurper().parseText(item)
+                issueJson << [docType: issueDocType]
+
+                logMessage += ' got result from API -> '
+                logMessage += elasticSearchHelper.updateItem("${issueDocType}_${issue.key}", JsonOutput.toJson(issueJson))
             } catch (Exception e) {
+                e.printStackTrace()
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/jira/issue/IssueIndexer.groovy]: Please use logger.

File: src/main/groovy/com/epam/esp/jira/issue/IssueIndexer.groovy [hunk 2]
```diff
@@ -95,9 +101,14 @@ class IssueIndexer implements Runnable {
...
                     }
                 }
-                logMessage += ' got result from API '
-                logMessage += elasticSearchHelper.updateItem(issue.key, item)
+
+                def issueJson = new JsonSlurper().parseText(item)
+                issueJson << [docType: issueDocType]
+
+                logMessage += ' got result from API -> '
+                logMessage += elasticSearchHelper.updateItem("${issueDocType}_${issue.key}", JsonOutput.toJson(issueJson))
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/jira/issue/IssueIndexer.groovy]: As long as we index projects, custom fields and issues we don't need to have ${issueDocType}_ prefix here.
Each ID will be unique anyway.

File: src/main/groovy/com/epam/esp/elasticsearch/comparator/TimestampIndexNameComparator.groovy [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package com.epam.esp.elasticsearch.comparator
+
+class TimestampIndexNameComparator implements Comparator<String> {
+
+    long offset
+    String delimiter
+
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/elasticsearch/comparator/TimestampIndexNameComparator.groovy:22]: By default the bith offset and delimiter are public. I don't think we need them public.

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy [hunk 4]
```diff
@@ -37,15 +42,86 @@ import org.joda.time.format.DateTimeFormat
...
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
+import java.util.Comparator
 
 import java.util.concurrent.ExecutionException
 import java.util.concurrent.TimeUnit
 
 class ElasticSearchHelper {
     final static Logger logger = LoggerFactory.getLogger(ElasticSearchHelper.class)
     String defaultIndex = null
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy]: "defaultIndex" ussage should be deprecated.

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy [hunk 5]
```diff
@@ -100,18 +188,29 @@ class ElasticSearchHelper {
...
+
     /**
      * JIRA 'issue' document specific
      * @param documentId
      * @param date
      * @return true if 'issue' update date is the same
      */
     boolean isInSync(String documentId, date) {
-        return isInSync(documentId, defaultType, date);
+        return isInSync(documentId, date);
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy]: Infinite recursion?

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy [hunk 6]
```diff
@@ -156,69 +272,59 @@ class ElasticSearchHelper {
...
-        }
-
-        nodes.each { key, value ->
-            client = client.addTransportAddress(new InetSocketTransportAddress(InetAddress.getByName(key), value))
-        }
+    String updateItem(String indexName, String documentId, String documentSrc) {
+        def jsonSlurper = new JsonSlurper()
+        def object = jsonSlurper.parseText(documentSrc)
+        def message = ''
+        def response = client.prepareIndex(indexName, defaultType, documentId).setSource(documentSrc, XContentType.JSON).get()
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy:215]: Please avoid document modification in helper class.

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy [hunk 7]
```diff
@@ -228,25 +334,61 @@ class ElasticSearchHelper {
...
         }
         logger.info("Alias $aliasName has been switched to index $indexName")
     }
+    /**
+    * Creates index
+    * @param name index name
+    * @param fieldsLimit total_fields.limit property
+    */
+    public void createIndex(String name, int fieldsLimit) {
+        CreateIndexResponse cir = client.admin().indices()
```
Comment (asamuta) [inline src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy:304]: Please rewrite it to accept parameters as map or remove this implementation from helper class.

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy (hunk 4 above)
Comment (maksim-lauryk) [reply src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy]: done

File: src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy (hunk 5 above)
Comment (maksim-lauryk) [reply src/main/groovy/com/epam/esp/elasticsearch/ElasticSearchHelper.groovy]: there is not recursion but method duplication was fixed

**Pull request merged**